controllers/managedocs_controller.go

```diff
@@ -825,8 +825,10 @@ func (r *ManagedOCSReconciler) reconcileOnboardinValidationSecret() error {
 		if err := r.own(r.onboardingValidationKeySecret); err != nil {
 			return err
 		}
+		r.addonParams[onboardingValidationKey] = strings.TrimSpace(r.addonParams[onboardingValidationKey])
+		onboardingValidationData := fmt.Sprintf("-----BEGIN PUBLIC KEY-----\n%s\n-----END PUBLIC KEY-----", r.addonParams[onboardingValidationKey])
```

Comment:
Do not mutate the actually provided value.

```suggestion
		onboardingValidationData := fmt.Sprintf(
			"-----BEGIN PUBLIC KEY-----\n%s\n-----END PUBLIC KEY-----", 												
			strings.TrimSpace(r.addonParams[onboardingValidationKey]),
		)
```